Lesson 08

Ya-Kate opened the pull request:
ДЗ урок № 8.

Commit:
Задания Урок № 7

Commit:
Сделала исправления по замечаниям, ДЗ урок №7.

Commit:
Выполнила обязательную часть ДЗ урок № 8.

artempolozov commented:
![image](https://user-images.githubusercontent.com/109097375/183460150-18e53ded-bfe3-4b67-a652-1afec724dd46.png)
опять файлы из двух ДЗ
проверю, когда исправишь

Commit:
Удален 7 урок, из 8 ветки.

artempolozov commented on src/lesson08/Circle.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.Objects;
+
+public class Circle implements Figure {
+    public String nameCircle = "Круг";
```
пока не советую к такому типу инициализации свойств объекта. лучше пока в конструкторе по умолчанию инициализировать данное свойство
не ошибка

artempolozov commented on src/lesson08/Circle.java:
```diff
@@ -0,0 +1,60 @@
+        return Objects.hash(nameCircle, lenghtCircl);
+    }
+
+    public Circle () { // конструктор по умолчанию
+
+    }
+
+    public Circle(int lenghtCircl) { // конструктор с параметрами
+        this.lenghtCircl = lenghtCircl;
+    }
```
непривычно видеть такую структуру. 
свойства, конструкторы, методы - как-то так выглядит класс

artempolozov reviewed (commented):
все таки по директория бы разбить. сложно разбирать просто список всех классов. минимум - разбивай по задания.

artempolozov commented on src/lesson08/Circle.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Override
+    public void squareFigure() { // переопределяется функция родительского класса  считается и выводится площадь круга
+        int sSquareFig = (int) (lenghtCircl * lenghtCircl / (4 * 3.14));
```
в классе Math есть соответствующая константа для числа Пи

artempolozov commented on src/lesson08/Triangle.java:
```diff
@@ -0,0 +1,86 @@
+    public Triangle () {  // конструктор по умолчанию
+
+    }
+    public Triangle(int sideOne, int sideTwo, int sideThree, int hFigure) { // конструктор с параметрами, инициализируются переменные данного класса
```
разве в треугольнике задается высота?

artempolozov commented on src/lesson08/Trapezoid.java:
```diff
@@ -0,0 +1,95 @@
+
+import java.util.Objects;
+
+public class Trapezoid implements Figure {
```
откуда трапеция взялась? в задании вроде не сказано

artempolozov commented on src/lesson08/HomeWorkPart1.java:
```diff
@@ -0,0 +1,29 @@
+Реализовать интерфейс с методом, который печатает название должности и
+имплементировать этот метод в созданные классы.
+         */
+        Figure triangle = new Triangle(10, 20, 30, 13);
```
откуда взялась высота?

artempolozov commented on src/lesson08/Accountant.java:
```diff
@@ -0,0 +1,44 @@
+        super(surname, name, salary);
+    }
+
+    public void PrintInfoAccountant() {
```
методы не именуются с большой буквы

artempolozov commented on src/lesson08/InfoPerson.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.util.Objects;
+
+public abstract class InfoPerson {
```
скорее так нужно было назвать интерфейс

artempolozov commented on src/lesson08/Employed.java:
```diff
@@ -0,0 +1,5 @@
+package lesson08;
+
+public interface Employed {
```
а так назвать абстрактный класс (точнее Employee)

artempolozov commented on src/lesson08/Director.java:
```diff
@@ -0,0 +1,44 @@
+        System.out.println("Должность: " + position);
+    }
+
+    public Director(String surname, String name, int salary) {
```
тогда уж и образование сюда впихивай

artempolozov commented on src/lesson08/Accountant.java:
```diff
@@ -0,0 +1,44 @@
+        System.out.println("Должность: " + position);
+    }
+
+    public Accountant(String surname, String name, int salary) {
```
то же самое с годовым балансом

artempolozov reviewed (approved):
все таки по директория бы разбить. сложно разбирать просто список всех классов. минимум - разбивай по задания. 
забыла о комментариях (комменты типа "конструктор по умолчанию" мало информативный)
если есть конструктор с параметрами, опиши параметры.
если есть класс, опиши и его

оценка 8 из 10

Commit:
Внесла корректировки урок № 8
в задании нужно массив сделать из 5 фигур,(сделала: круг, треугольник, квадрат, прямоугольник, трапеция.
